### ISSUE TITLE
Generate roadside parking for a given network

### ISSUE THREAD

**Issue by m-kro** [eclipse/sumo#13563]
Context / origin: Distribute charging stations on existing parking spaces. Import from OSM #3602 is still missing and OSM parking data is very incomplete. Thus a script to add some roadside parking to a selection of edges with some heuristic / hints would be a nice idea to create urban network with much roadside parking.

**Comment by namdre** [comment]
how about https://sumo.dlr.de/docs/Tools/Misc.html#generateparkingareaspy

**Comment by m-kro** [comment]
OK this can be a starting point. Maybe there should be some options to control how random the parking creation should be:

- apply only to selection of edges (such that other parts of the network can processed in a separate run with other parameters)
- check if there is an opposite edge when putting parking on the left side
- optionally limit capacity to the space actually available
- configure onRoad attribute

**Comment by namdre** [comment]
The new code adds the lane index to the parking area id even though the tool still creates at most one parking area per edge. Is there a good reason for adding the index? (at the moment, the modified ids are making the tests fail).

**Comment by m-kro** [comment]
Yes I'll update the tests and add new one(s) to show the new features. For on-road parking, there is the option to use either a specific lane or all.
